lib/gooddata/lcm/helpers/check_helper.rb

```diff
@@ -14,9 +14,10 @@ def check_params(specification, params)
           specification.keys.each do |param_name|
             value = params.send(param_name)
             type = specification[param_name][:type]
-
-            if value.nil?
-              if specification[param_name][:opts][:required]
+            if value.nil? or ( value.is_a? String and value.empty? )
+              if specification[param_name][:opts][:default]
```

Comment:
Refers to 'specification' more than self (maybe move it to another class?) (FeatureEnvy)